reVX/handlers/sc_points.py

```diff
@@ -0,0 +1,631 @@
+
+        Returns
+        -------
+        out : tuple
+            (sc_point, capacity, availability)
+        """
+        build_capacity = capacity.copy()
+        if self.capacity > 0:
+            if capacity < self.capacity:
+                drop = 0
+                for cap in self.available_capacities:
+                    drop += 1
+                    capacity -= cap
+                    if capacity <= 0:
+                        break
+            else:
+                drop = None
+
+            out = self._drop_build_capacity(build_capacity, drop=drop)
+        else:
+            msg = "{} has no remaining capacity".format(self)
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        return out
+
+    @classmethod
+    def create(cls, sc_point, res_cf_means):
+        """
+        Create Point from supply curve point meta and resource meta
+
+        Parameters
+        ----------
+        sc_point : pandas.Series
+            Supply curve point meta data
+        res_cf_means : pandas.Series
+            Resource cf_means by gid
+
+        Returns
+        -------
+        Point
+            Initialized Point objects
+        """
+        sc_gid = sc_point.name
+        capacity = sc_point['capacity']
+        res_gids = sc_point['res_gids']
+        gid_counts = sc_point['gid_counts']
+        res_cf = res_cf_means.loc[res_gids].values
+
+        return cls(sc_gid, capacity, res_gids, gid_counts, res_cf)
+
+
+class SupplyCurvePoints:
+    """
+    Class to handle Supply Curve points and their meta
+    """
+
+    def __init__(self, sc_table, res_meta, max_workers=None,
+                 points_per_worker=400, offshore=False):
+        """
+        Parameters
+        ----------
+        sc_table : str | pandas.DataFrame
+            Supply Curve table .csv or pre-loaded pandas DataFrame
+        res_meta : str | pandas.DataFrame
+            Path to resource .h5, generation .h5, or pre-extracted .csv or
+            pandas DataFrame
+        max_workers : int, optional
+            Number of workers to use for point creation, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+        points_per_worker : int, optional
+            Number of points to create on each worker, by default 400
+        """
+        self._sc_table = self._parse_sc_table(sc_table, offshore=offshore)
+
+        self._sc_points, self._capacity, self._mask = \
+            self._parse_sc_points(self._sc_table, res_meta,
+                                  max_workers=max_workers,
+                                  points_per_worker=points_per_worker,
+                                  offshore=offshore)
+
+    def __repr__(self):
+        msg = "{} with {} points".format(self.__class__.__name__, len(self))
+        return msg
+
+    def __len__(self):
+        return len(self._sc_points)
+
+    def __getitem__(self, sc_gid):
```

Comment:
I really think the getters and setters should have docstrings. Just because pylint doesnt require it doesnt mean its not important. What does getting an sc_gid from this object return? No idea.

Reply:
Fair enough, added